src/camera_capture/camera_capture/camera_capture.py

```diff
@@ -0,0 +1,206 @@
+        self.get_logger().debug(
+            f"Incomming message on topic livestream \nwith message: {topic_msg}"
+        )
+        timer_period = 0.03  # seconds TODO: make into settings 30hz
+        if self.cap.isOpened():
+            self.get_logger().info("camera is available")
+            if topic_msg.data:
+                self.get_logger().info("starting livestream")
+                timer = self.create_timer(timer_period, self.timer_callback)
+            elif not topic_msg.data:
+                self.get_logger().info("stopping livestream")
+                try:
+                    timer.cancel()
+                    timer.destroy()
+                except UnboundLocalError:
+                    self.get_logger().error("timer was is not defined)")
+            else:
+                self.get_logger().debug(
+                    "can't stop livestream because livestream is already stopped"
+                )
+        elif not self.cap.isOpened():
+            self.get_logger().info("camera not available")
+
+    def timer_callback(self):
+        """callback function to read image data from camera and publish it to the topic of the livestream"""
+        ret, frame = self.cap.read()
+        msg_image = self.bridge.cv2_to_imgmsg(np.array(frame), "bgr8")
+        self.pub_cam_livestream.publish(msg_image)
+
+    def capture_snapshot_callback(self, topic_msg: Bool):
+        """Captures images when a True is received on the snapshot topic and publishes the immage on a topic.
+
+        Args:
+            topic_msg (Bool): True for capturing snapshot, False is not used.
+        """
+        self.get_logger().info(
+            f"message received on topic snapshot \nwith message: {topic_msg}\nwith data :{topic_msg.data}"
+        )
+        self.cap.set(
+            cv2.CAP_PROP_FRAME_HEIGHT, json_settings["capture_height_snapshot"]
+        )
+        self.cap.set(cv2.CAP_PROP_FRAME_WIDTH, json_settings["capture_width_snapshot"])
+        if self.cap.isOpened():
+            ret, frame = self.cap.read()
+            print(type(frame))
+            msg_image = self.bridge.cv2_to_imgmsg(np.array(frame), "bgr8")
+            msg_image.header.frame_id = str(self.image_counter)
+            cv2.imwrite(f"{self.image_location}/image{self.image_counter}.jpg", frame)
+            self.get_logger().info(f"image saved at image{self.image_counter}.jpg")
+            self.pub_cam_snapshot.publish(msg_image)
+            self.image_counter += 1
+            prevent_overflood_image(self.image_counter)
+        else:
+            self.get_logger().info("camera not available")
+        self.cap.set(
+            cv2.CAP_PROP_FRAME_HEIGHT, json_settings["capture_height_livestream"]
+        )
+        self.cap.set(
+            cv2.CAP_PROP_FRAME_WIDTH, json_settings["capture_width_livestream"]
+        )
+
+
+def prevent_overflood_image(img_number: int):
+    """makes sure that there are only 5 images in the image folder
```

Comment:
Just a comment, please do not take the time to change this for this PR. But it is good practice to write the first line of your doc string in a uniform manner. A common pattern is an imperative phrase (e.g. "Make sure that there are only 5 images in the image folder").

Right now, there are multiple styles in this document, making reading them ever so slightly more difficult